planner/core/exhaust_physical_plans.go

```diff
@@ -1969,61 +1969,75 @@ func filterIndexJoinBySessionVars(sc sessionctx.Context, indexJoins []PhysicalPl
 	return indexJoins
 }
 
+func (p *LogicalJoin) prefer(joinFlags ...uint) bool {
+	for _, flag := range joinFlags {
+		if p.preferJoinType&flag > 0 {
+			return true
+		}
+	}
+	return false
+}
+
+// canPassJoinHint returns whether this join plan can pass current join hints.
+func (p *LogicalJoin) canPassJoinHint(join PhysicalPlan, leftOuter bool) bool {
```

Comment:
Suggest renaming it like `indexJoinSatisfiesHint` because the actual logic assumes `join` is an index join.

Reply:
We can also use `(*basePhysicalJoin).getInnerChildIdx()` to avoid the need to pass the `leftOuter`.

Reply:
Good suggestion, updated.